main.py

```diff
@@ -105,11 +105,14 @@ def get_next_location(distances_df, km_left):
         apply_distance_calc, args=(start_lat, start_lon), axis=1
     )
     distances_df["ratio"] = distances_df.apply(calculate_ratio, axis=1)
-    best_next_location = pd.Series(distances_df.query("distance < 1000").ratio.idxmax())
-    if distances_df.distance_to_home.loc[best_next_location].values[0] > km_left:
+    best_next_location = distances_df.query("distance < 1000").ratio.idxmax()
```

Comment:
You dont need to do distance < 1000 as you are doing another check against distance two lines later.
Please remove it or explain what are the benefits of having this line.

Reply:
Ratio is calculated without max distance in mind, so best ratio brewery might be too far away to have enough fuel to come back, hence the query. 
Fixed: I moved this query to an earlier step, outside of this function.